spec/features/step_nav_helper_spec.rb

```diff
@@ -0,0 +1,185 @@
+
+      visit '/step-nav/vomit-comet'
+    end
+
+    it 'does not show step nav related components' do
+      expect(page).not_to have_selector('.gem-c-step-nav-header')
+      expect(page).not_to have_selector('.gem-c-step-nav-related')
+      expect(page).not_to have_selector('.gem-c-step-nav')
+    end
+  end
+
+  context 'one related step by step navigation journey' do
+    before do
+      content_store_has_random_item(base_path: '/vomit-comet', schema: 'transaction', part_of_step_navs: [spacewalk_step_nav])
+
+      visit '/step-nav/vomit-comet'
+    end
+
+    it 'shows step nav related links' do
+      expect(page).to have_selector('.gem-c-step-nav-related')
+
+      within('.gem-c-step-nav-related') do
+        expect(page).to have_selector('.gem-c-step-nav-related__link', count: 1)
+        expect(page).to have_link('Learn to spacewalk: small step by giant leap', href: '/learn-to-spacewalk')
+      end
+    end
+
+    it 'shows the full step nav sidebar' do
+      within('.gem-c-step-nav') do
+        expect(page).to have_css('h3', count: 3)
+        expect(page).to have_css('.gem-c-step-nav__link--active', count: 1, text: 'Experience zero gravity in the atmosphere') # one active link
+      end
+    end
+
+    it 'shows the step nav header' do
+      within('.gem-c-step-nav-header') do
+        expect(page).to have_link('Learn to spacewalk: small step by giant leap', href: '/learn-to-spacewalk')
+      end
+    end
+  end
+
+  context 'multiple related step by step navigation journeys' do
+    before do
+      content_store_has_random_item(base_path: '/vomit-comet', schema: 'transaction', part_of_step_navs: [
+        {
+          "content_id" => "8f5d4f2b-daf0-4460-88c1-fdd76c90f6f1",
+          "locale" => "en",
+          "title" => "Learn to spacewalk: small step by giant leap",
+          "base_path" => "/learn-to-spacewalk"
+        },
+        {
+          "content_id" => "8f5d4f2b-daf0-4460-88c1-fdd76c90f6f2",
+          "locale" => "en",
+          "title" => "Lose your lunch: lurch by lurch",
+          "base_path" => "/lose-your-lunch"
+        }
+      ])
+
+      visit '/step-nav/vomit-comet'
+    end
+
+    it 'shows step nav related links' do
+      expect(page).to have_selector('.gem-c-step-nav-related')
+
+      within('.gem-c-step-nav-related') do
+        expect(page).to have_selector('.gem-c-step-nav-related__link-item', count: 2)
+        expect(page).to have_link('Learn to spacewalk: small step by giant leap', href: '/learn-to-spacewalk')
+        expect(page).to have_link('Lose your lunch: lurch by lurch', href: '/lose-your-lunch')
+      end
+    end
+
+    it "doesn't show the full step nav sidebar or header" do
+      expect(page).not_to have_selector('.gem-c-step-nav-header')
+      expect(page).not_to have_selector('.gem-c-step-nav')
+    end
+  end
+
+
+  def content_store_has_random_item(base_path:, schema: 'placeholder', part_of_step_navs: [])
```

Comment:
Maybe we should add a note to extract this into `gds-api-adapters` as `frontend` also has this method (just not the same exact code)

Reply:
👍